mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -443,6 +444,144 @@ class IntegerOpConversion final : public OpConversionPattern<ArithOp> {
   }
 };
 
+template <typename ArithOp, typename EmitCOp>
+class BitwiseOpConversion : public OpConversionPattern<ArithOp> {
+public:
+  using OpConversionPattern<ArithOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ArithOp op, typename ArithOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Type type = this->getTypeConverter()->convertType(op.getType());
+    if (!isa_and_nonnull<IntegerType, emitc::SignedSizeTType, emitc::SizeTType>(
+            type)) {
+      return rewriter.notifyMatchFailure(
+          op, "expected integer or size_t/ssize_t type");
+    }
+
+    // There is no unsigned i1 type, bitwise ops can be performed directly
```

Comment:
I don't understand this comment. The type `ui1` seems to exist.

Reply:
I had support for ui1s until I read somewhere in the codebase that this type was illegal, and I then removed that code. I can't find that place anymore, plus ui1 indeed seems to exist. 

Nevertheless, we don't need to perform the conversion here anyway for i1s, so the comment can go and the behavior stay.